src/dmd/dtoh.d

```diff
@@ -196,97 +230,125 @@ extern(C++) void genCppHdrFiles(ref Modules ms)
+        HashIf(buf, "!defined(BEGIN_ENUM_TYPE)");
+        {
+            HashDefine(buf, "BEGIN_ENUM_TYPE(type, name, upper, lower) namespace name {");
+            HashDefine(buf, "ENUM_KEY_TYPE(type, name, value, enumName, upper, lower, abbrev) static type const name = value;");
+            HashDefine(buf, "END_ENUM_TYPE(type, name, upper, lower) };");
+        }
+        HashEndIf(buf);
     }
     if (v.hasAnonEnum)
     {
-        buf.writestring("#if !defined(BEGIN_ANON_ENUM)\n");
-        buf.writestring("# define BEGIN_ANON_ENUM() enum {\n");
-        buf.writestring("# define ANON_ENUM_KEY(type, name, value) name = value,\n");
-        buf.writestring("# define END_ANON_ENUM() };\n");
-        buf.writestring("#endif\n");
+        HashIf(buf, "!defined(BEGIN_ANON_ENUM)");
+        {
+            HashDefine(buf, "BEGIN_ANON_ENUM() enum {");
+            HashDefine(buf, "ANON_ENUM_KEY(type, name, value) name = value,");
+            HashDefine(buf, "END_ANON_ENUM() };");
+        }
+        HashEndIf(buf);
     }
     if (v.hasAnonNumericEnum)
     {
-        buf.writestring("#if !defined(BEGIN_ANON_ENUM_NUMERIC)\n");
-        buf.writestring("# define BEGIN_ANON_ENUM_NUMERIC(type) enum : type {\n");
-        buf.writestring("# define ANON_ENUM_KEY_NUMERIC(type, name, value) name = value,\n");
-        buf.writestring("# define END_ANON_ENUM_NUMERIC(type) };\n");
-        buf.writestring("#endif\n");
+        HashIf(buf, "!defined(BEGIN_ANON_ENUM_NUMERIC)");
+        {
+            HashDefine(buf, "BEGIN_ANON_ENUM_NUMERIC(type) enum : type {");
+            HashDefine(buf, "ANON_ENUM_KEY_NUMERIC(type, name, value) name = value,");
+            HashDefine(buf, "END_ANON_ENUM_NUMERIC(type) };");
+        }
+        HashEndIf(buf);
     }
     if (v.hasNumericConstant)
     {
-        buf.writestring("#if !defined(ENUM_CONSTANT_NUMERIC)\n");
-        buf.writestring("# define ENUM_CONSTANT_NUMERIC(type, name, value) enum : type { name = value };\n");
-        buf.writestring("#endif\n");
+        HashIf(buf, "!defined(ENUM_CONSTANT_NUMERIC)");
+        {
+            HashDefine(buf, "ENUM_CONSTANT_NUMERIC(type, name, value) enum : type { name = value };");
+        }
+        HashEndIf(buf);
     }
     if (v.hasTypedConstant)
     {
-        buf.writestring("#if !defined(ENUM_CONSTANT)\n");
-        buf.writestring("# define ENUM_CONSTANT(type, name, value) static type const name = value;\n");
-        buf.writestring("#endif\n");
+        HashIf(buf, "!defined(ENUM_CONSTANT)");
+        {
+            HashDefine(buf, "ENUM_CONSTANT(type, name, value) static type const name = value;");
+        }
+        HashEndIf(buf);
     }
-    buf.writestring("\n");
+    buf.writenl();
 
     buf.write(&fwd);
     if (fwd.length > 0)
-        buf.writestring("\n");
+        buf.writenl();
 
     buf.write(&done);
     buf.write(&decl);
 
     debug (Debug_DtoH)
     {
-        buf.writestring(`
-#if OFFSETS
+        buf.writestring(`#if OFFSETS
```

Comment:
Why not `hashIf`?

Reply:
just because I didn't want to touch this code block which is only in debug builds